internal/model/puller.go

```diff
@@ -791,7 +791,7 @@ func (p *Puller) copierRoutine(in <-chan copyBlocksState, pullChan chan<- pullBl
 			p.progressEmitter.Register(state.sharedPullerState)
 		}
 
-		dstFd, err := state.tempFile()
+		dstFd, err := state.tempFile(p.ignorePerms)
```

Comment:
I think this would be cleaner as a field on the `sharedPullerState` directly, set at creation time.